enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,504 @@
+
+Namespace Label Management: Admins seeking to implement this proposal need to introduce tools, scripts or organizational processes to assist administrators in managing labels on namespaces that should be exempt from the default scheduling restrictions.
+This could include automation for emergency situations where rapid response is necessary.
+
+### Expected Outcomes
+
+- Improved protection of critical cluster resources, ensuring that control plane nodes and nodes with
+  specialized resources are shielded from inappropriate workloads.
+- Increased flexibility for administrators to tailor the enforcement scheduling policies to the specific needs of their organization and operational environment.
+
+### Workflow Description
+
+<!-- Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials. -->
+
+This section outlines the workflow for enforcing the proposed mechanism for prevention of non-control plane workloads from being scheduled on control plane nodes and allow for exceptions based on administrative configurations.
+
+#### Actors
+
+Pre-Workflow:
+
+- OpenShift Engineers: Ensure all control plane and essential workloads ship with the correct tolerations and permissions out of the box to seamlessly operate on NoExecute tainted nodes. This preparation is crucial for the uninterrupted functioning of OpenShift's core services.
+
+Workflow:
+
+- Cluster Administrator: Tasked with applying NoExecute taints to control plane nodes or nodes belonging to other specialized groups.
+For extending this approach to other special node groups, they must coordinate with relevant teams to ensure those workloads include necessary tolerations and permissions.
+They are also responsible for creating and applying the ValidatingAdmissionPolicyBinding (that will enable the enforcement of the provided ValidatingAdmissionPolicy we ship) resources to configure which taint tolerations are disallowed for pods.
```

Comment:
Actually, we can create the ValidatingAdmissionPolicyBinding too and leave it with MatchPolicy of Audit and Warn, right?  Then to have a "normal" configuration, the cluster-admin can simply add `Deny`.